src/gui/categoryoptionsdialog.cpp

```diff
@@ -0,0 +1,158 @@
+    ui->sequentialOrderBox->setChecked(BitTorrent::Session::instance()->isCategoryDownloadSequential(m_category));
+    ui->bandwidthBox->setChecked(BitTorrent::Session::instance()->isCategoryBandwidthSettingsEnabled(m_category));
+
+    int maxDownloadSlider = Preferences::instance()->getGlobalDownloadLimit() / 1024;
+    int downloadLimitInKb = BitTorrent::Session::instance()->getCategoryDownloadLimit(m_category) / 1024;
+    int upload_limit_in_kb = BitTorrent::Session::instance()->getCategoryUploadLimit(m_category) / 1024;
+    if (maxDownloadSlider < 1000)
+        maxDownloadSlider = 1000;
+    if (maxDownloadSlider < downloadLimitInKb)
+        maxDownloadSlider = downloadLimitInKb;
+
+    int maxUploadSlider = Preferences::instance()->getGlobalUploadLimit() / 1024;
+    if (maxUploadSlider < 1000)
+        maxUploadSlider = 1000;
+    if (maxUploadSlider < upload_limit_in_kb)
+        maxUploadSlider = upload_limit_in_kb;
+
+    ui->downloadSlider->setMaximum(maxDownloadSlider);
+    ui->uploadSlider->setMaximum(maxUploadSlider);
+    ui->downloadSlider->setValue(downloadLimitInKb);
+    ui->uploadSlider->setValue(upload_limit_in_kb);
+    CategoryOptionsDialog::updateSpinValue(ui->spinDownload, downloadLimitInKb);
+    CategoryOptionsDialog::updateSpinValue(ui->spinUpload, upload_limit_in_kb);
+
+    ui->ratioBox->setChecked(BitTorrent::Session::instance()->isCategoryRatioSettingsEnabled(m_category));
+    ui->ratioSpin->setEnabled(BitTorrent::Session::instance()->getCategoryRatioLimitType(m_category) == 3);
+    ui->ratioSpin->setMinimum(0);
+    ui->ratioSpin->setMaximum(BitTorrent::TorrentHandle::MAX_RATIO);
+    ui->ratioSpin->setValue(BitTorrent::Session::instance()->getCategoryRatioLimit(m_category));
+    ui->ratioButtonGroup->setId(ui->globalRatioLimitRadio, 0);
+    ui->ratioButtonGroup->setId(ui->noRatioLimitRadio, 1);
+    ui->ratioButtonGroup->setId(ui->specificRatioLimitRadio, 2);
+    ui->ratioButtonGroup->button(m_ratioLimitType)->setChecked(true);
+
+    connect(ui->downloadSlider, SIGNAL(valueChanged(int)), this, SLOT(updateDownloadSpinValue(int)));
+    connect(ui->spinDownload, SIGNAL(valueChanged(int)), this, SLOT(updateDownloadSliderValue(int)));
+    connect(ui->uploadSlider, SIGNAL(valueChanged(int)), this, SLOT(updateUploadSpinValue(int)));
+    connect(ui->spinUpload, SIGNAL(valueChanged(int)), this, SLOT(updateUploadSliderValue(int)));
+    connect(ui->ratioButtonGroup, SIGNAL(buttonClicked(int)), this, SLOT(ratioLimitTypeChanged(int)));
+}
+
+CategoryOptionsDialog::~CategoryOptionsDialog()
+{
+    delete ui;
+}
+
+void CategoryOptionsDialog::on_browseButton_clicked()
+{
+    QDir pathDir(m_path);
+    QString dir;
+    if (!m_path.isEmpty() && pathDir.exists())
+        dir = QFileDialog::getExistingDirectory(this, tr("Choose a download directory"), pathDir.absolutePath());
+    else
+        dir = QFileDialog::getExistingDirectory(this, tr("Choose a download directory"), QDir::homePath());
```

Comment:
You do not need to repeat the `QFileDialog::getExistingDirectory()` call just to pass one different parameter.